client/app/private-database/configuration/private-database-configuration.controller.js

```diff
@@ -24,41 +25,73 @@ angular.module("App").controller(
 
         getConfigurationDetails () {
             this.loading = true;
-            this.privateDatabaseService
+            return this.privateDatabaseService
                 .getConfigurationDetails(this.productId)
                 .then((config) => {
-                    this.configurations = config.details;
-                    this.baseConfiguration = angular.copy(config.details);
+                    this.configurations = config.details.map((field) => this.convertAvailableValues(field));
                 })
-                .catch((err) => {
-                    this.alerter.alertFromSWS(this.$scope.tr("privateDatabase_configuration_error"), err, this.$scope.alerts);
+                .catch((error) => {
+                    this.alerter.error(this.translator.tr("privateDatabase_configuration_error"), this.$scope.alerts.main);
+                    this.$q.reject(error);
                 })
                 .finally(() => {
                     this.loading = false;
                 });
         }
 
+        convertAvailableValues (field) {
+            field.description = this.getFieldDescriptionTranslated(field);
+            field.availableValues = field.availableValues.map((value) => ({ id: value, text: value }));
+            if (field.key === "autocommit") {
+                PrivateDatabaseConfigurationsCtrl.convertFieldAsToggle(field);
+                field.availableValues = [{ id: "0", text: " OFF" }, { id: "1", text: "ON" }];
+            } else if (field.key === "event_scheduler") {
+                PrivateDatabaseConfigurationsCtrl.convertFieldAsToggle(field);
+            } else {
+                field.type = "select";
+                field.selectedValue = field.availableValues.find((value) => value.id === field.value);
+            }
+            return field;
+        }
+
+        static convertFieldAsToggle (field) {
+            field.type = "toggle";
+            field.selectedValue = { id: field.value };
```

Comment:
I now make a distinction between the field_id and the field_text value, I avoid using ```value``` to make it clear. The view have this model to work with```{ id: value, text: value }```, instead of only a string.